executor/common_linux.h

```diff
@@ -1570,6 +1570,310 @@ static long syz_init_net_socket(volatile long domain, volatile long type, volati
+		return;
+	}
+	case HCI_OP_READ_BUFFER_SIZE: {
+		struct hci_rp_read_buffer_size rp;
+		rp.status = 0;
+		rp.acl_mtu = 1021;
+		rp.sco_mtu = 96;
+		rp.acl_max_pkt = 4;
+		rp.sco_max_pkt = 6;
+		hci_send_event_cmd_complete(fd, hdr->opcode, &rp, sizeof(rp));
+		return;
+	}
+	}
+
+	char status[0xf9] = {0};
+	hci_send_event_cmd_complete(fd, hdr->opcode, status, sizeof(status));
+}
+
+static void* event_thread(void* arg)
+{
+	int epoll_fd = *(int*)arg;
+
+	while (!vhci_init) {
+		struct epoll_event event;
+		epoll_pwait(epoll_fd, &event, 1, -1, NULL);
+
+		char buf[BUF_SIZE];
+		size_t buf_size = read(event.data.fd, buf, sizeof(buf));
+		if (buf[0] == HCI_COMMAND_PKT)
+			process_command_pkt(event.data.fd, buf, buf_size);
+	}
+
+	return NULL;
+}
+
+static void initialize_vhci()
+{
+#if SYZ_EXECUTOR
+	if (!flag_vhci_injection)
+		return;
+#endif
+
+	vhci_fd = open("/dev/vhci", O_RDWR);
+	if (vhci_fd == -1)
+		fail("open /dev/vhci failed");
+
+	int hci_socket = syz_init_net_socket(AF_BLUETOOTH, SOCK_RAW, 1);
+	if (hci_socket < 0)
+		fail("syz_init_net_socket failed");
+
+	int epoll_fd = epoll_create1(0);
+	if (epoll_fd == -1)
+		fail("epoll_create1 failed");
+
+	struct epoll_event event = {0};
+	event.events = EPOLLIN;
+	event.data.fd = vhci_fd;
+	if (epoll_ctl(epoll_fd, EPOLL_CTL_ADD, vhci_fd, &event))
+		fail("epoll_ctl failed");
+
+	pthread_t th;
+	if (pthread_create(&th, NULL, event_thread, &epoll_fd))
+		fail("pthread_create failed");
+
+	for (int i = 0; i < 1000; i++) {
+		struct hci_dev_info info = {0};
+		info.dev_id = HCI_DEV;
+		ioctl(hci_socket, HCIGETDEVINFO, &info);
+		if (info.flags & 1) // UP
+			break;
+		usleep(1000);
+	}
+
+	struct hci_dev_req dr = {0};
+	dr.dev_id = HCI_DEV;
+	dr.dev_opt = SCAN_PAGE;
+	if (ioctl(hci_socket, HCISETSCAN, &dr) < 0)
+		fail("ioctl(HCISETSCAN) failed");
+
+	struct hci_ev_conn_request request = {0};
+	memcpy(&request.bdaddr, "\x66\x55\x44\x33\x22\x11", 6);
+	request.link_type = ACL_LINK;
+	hci_send_event_packet(vhci_fd, HCI_EV_CONN_REQUEST, &request, sizeof(request));
+
+	struct hci_ev_conn_complete complete = {0};
+	complete.status = 0;
+	complete.handle = ACL_HANDLE;
+	memcpy(&complete.bdaddr, "\x66\x55\x44\x33\x22\x11", 6);
+	complete.link_type = ACL_LINK;
+	complete.encr_mode = 0;
+	hci_send_event_packet(vhci_fd, HCI_EV_CONN_COMPLETE, &complete, sizeof(complete));
+
+	struct hci_ev_remote_features features = {0};
+	features.status = 0;
+	features.handle = ACL_HANDLE;
+	hci_send_event_packet(vhci_fd, HCI_EV_REMOTE_FEATURES, &features, sizeof(features));
+
+	close(hci_socket);
+
+	vhci_init = 1;
+	pthread_kill(th, SIGKILL);
+	pthread_join(th, NULL);
+}
+#endif
+
+#if SYZ_EXECUTOR || __NR_syz_emit_vhci
+#if SYZ_EXECUTOR || SYZ_VHCI_INJECTION
+long syz_emit_vhci(volatile long a0, volatile long a1)
+{
+	if (vhci_fd < 0)
+		return (uintptr_t)-1;
+
+	char* data = (char*)a0;
+	uint32 length = a1;
+
+	return write(vhci_fd, data, length);
+}
+#else
+long syz_emit_vhci(volatile long a0, volatile long a1)
```

Comment:
We don't have this for a similar syz_emit_ethernet:
```
#if SYZ_EXECUTOR || __NR_syz_emit_ethernet && SYZ_NET_INJECTION
static long syz_emit_ethernet(volatile long a0, volatile long a1, volatile long a2)
```

What happens if you remove this empty function? Is there a reason to diverge from what we do for syz_emit_ethernet? If not, then please do the same for consistency.